zebra-chain/src/sprout/tree.rs

```diff
@@ -10,11 +10,71 @@
 //!
 //! A root of a note commitment tree is associated with each treestate.
 #![allow(clippy::unit_arg)]
+#![allow(dead_code)]
 
-use std::fmt;
+use std::{collections::VecDeque, fmt};
 
+use byteorder::{BigEndian, ByteOrder};
+use lazy_static::lazy_static;
 #[cfg(any(test, feature = "proptest-impl"))]
 use proptest_derive::Arbitrary;
+use sha2::digest::generic_array::GenericArray;
+
+use super::commitment::NoteCommitment;
+
+const MERKLE_DEPTH: usize = 29;
+
+/// MerkleCRH^Sprout Hash Function
+///
+/// Used to hash incremental Merkle tree hash values for Sprout.
+///
+/// MerkleCRH^Sprout(layer, left, right) := SHA256Compress(left || right)
+///
+/// `layer` is unused for Sprout but used for the Sapling equivalent.
+///
+/// https://zips.z.cash/protocol/protocol.pdf#merklecrh
+fn merkle_crh_sprout(left: [u8; 32], right: [u8; 32]) -> [u8; 32] {
+    let mut other_block = [0u8; 64];
+    other_block[..32].copy_from_slice(&left[..]);
+    other_block[32..].copy_from_slice(&right[..]);
+
+    // H256: Sha256 initial state
+    // https://github.com/RustCrypto/hashes/blob/master/sha2/src/consts.rs#L170
+    let mut state = [
+        0x6a09e667, 0xbb67ae85, 0x3c6ef372, 0xa54ff53a, 0x510e527f, 0x9b05688c, 0x1f83d9ab,
+        0x5be0cd19,
+    ];
```

Comment:
It's sad we can't just get these constants from the library itself.

Reply:
💯